lib/libzfs/libzfs_sendrecv.c

```diff
@@ -3755,7 +3887,8 @@ zfs_receive_impl(libzfs_handle_t *hdl, const char *tosnap,
 		assert(DMU_GET_STREAM_HDRTYPE(drrb->drr_versioninfo) ==
 		    DMU_COMPOUNDSTREAM);
 		return (zfs_receive_package(hdl, infd, tosnap, flags, &drr,
```

Comment:
This was not intentional and i will try to fix it: it's my first time reading zfs_receive_*() functions and it was kind of confusing trying to understand the code path of a `zfs recv` ... sorry about that.